src/lib/statistics/statistics_objects/histogram_domain.hpp

```diff
@@ -0,0 +1,105 @@
+#pragma once
+
+#include <cmath>
+#include <limits>
+#include <string>
+
+#include "types.hpp"
+
+namespace opossum {
+
+/**
+ * HistogramDomain<T> is a template specialized for ints, floats, and strings respectively. It provides a function
+ * `next_value()` for all three type categories and `previous_value()` for ints and floats. This enables algorithms
+ * working on histograms to be written more generically than having to specialize templates all over the place.
+ *
+ * The reason this is a class template, as opposed to more simple functions, is that the
+ * HistogramDomain<std::string> has a state (a character range and a prefix length)
+ */
+
+template <typename T, typename Enable = void>
+class HistogramDomain {};
+
+template <typename T>
+class HistogramDomain<T, std::enable_if_t<std::is_integral_v<T>>> {
+ public:
+  T next_value(T v) const {
+    if (v == std::numeric_limits<T>::max()) return v;
```

Comment:
Renamed to `next_value_clamped` and doc'ed with 

```c++
 * HistogramDomain<T> is a template specialized for integral types, floating point types, and strings respectively.
 * It provides a function `next_value_clamped()` for all three type categories and `previous_value_clamped()`
 * for arithmetic types.
 * The functions are "clamped" simply because, e.g. there is no next value for INT_MAX in the domain of integers. As
 * histograms are only used for estimations (and not for pruning decisions), this is acceptable.
```